Sources/NIO/FixedLengthFrameDecoder.swift

```diff
@@ -0,0 +1,47 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public final class FixedLengthFrameDecoder<T: FixedWidthInteger>: ByteToMessageDecoder {
```

Comment:
I think we should rename this to `LengthFieldBasedFrameDecoder` as its not really fixes but depends on the length that is used to prefix the actual data.